lib/VisualGenerator.js

```diff
@@ -89,7 +89,104 @@ function copyVisualTemplate(targetPath, templateName) {
+                VisualSettingsMethods += '      public ' + settingsSubClass + ': ' +
+                    settingsSubClass + 'Settings = new ' + settingsSubClass + 'Settings();\n';
+
+                let settingsSubClassDefinition = '\n    export class ' + settingsSubClass + 'Settings {\n';
+
+                if (subClassProperties) {
+                    for (let property in subClassProperties) {
+                        if (verbose) {
+                            console.log('property: ' + subClassProperties[property].displayName);
+                        }
+                        let propertyType = subClassProperties[property].type;
+                        if (verbose) {
+                            console.log('prop type: ' + propertyType);
+                        }
+                        // add some spaghetti to this meals
+                        settingsSubClassDefinition += '     // ' + subClassProperties[property].displayName +
+                            '\n      public ' + property;
 
+                        if (verbose) {
+                            console.log(propertyType.keys);
+                        }
+                        if (propertyType.bool) {
+                            if (verbose) {
+                                console.log('bool');
+                            }
+                            settingsSubClassDefinition += ': boolean = true;\n';
+                        } else
+                        if (propertyType.fill) {
+                            
+                            if (verbose) {
+                                console.log('fill');
+                            }
+                            settingsSubClassDefinition += ': string = "";\n';
+                        } else
+                        if (propertyType.formatting) {
+                            if (verbose) {
+                                console.log('formatting');
+                            }
+                            if (propertyType.formatting.fontSize) {
+                                settingsSubClassDefinition += ': number = 12;\n';
+                            }
+                        }
+                    }
+                }
+                settingsSubClassDefinition += '     }\n';
+                subClassesDefinitionArray.push(settingsSubClassDefinition);
+            }
+            VisualSettingsMethods += '      }\n';
+            // Add properties class per each setting
+            for (let subDefinitions in subClassesDefinitionArray) {
+                VisualSettingsMethods += subClassesDefinitionArray[subDefinitions];
+            }
+            if (verbose) {
+                console.log('final text: ' + VisualSettingsMethods);
+            }
+
+        } else {
+            if (verbose) {
+                console.log('Settings not found');
+            }
+            VisualSettingsMethods += '// TODO: fill all visual settings here\n ' +
```

Comment:
this should also be on our list: default values directly from capabilities.
Also in separate iteration